rasa/core/domain.py

```diff
@@ -265,30 +265,61 @@ def collect_slots(slot_dict: Dict[Text, Any]) -> List[Slot]:
 
     @staticmethod
     def collect_intent_properties(
-        intents: List[Union[Text, Dict[Text, Any]]]
+        intents: List[Union[Text, Dict[Text, Any]]], entities: List[Text]
     ) -> Dict[Text, Dict[Text, Union[bool, List]]]:
         intent_properties = {}
         for intent in intents:
-            if isinstance(intent, dict):
-                name = list(intent.keys())[0]
-                for properties in intent.values():
-                    properties.setdefault("use_entities", True)
-                    properties.setdefault("ignore_entities", [])
-                    if (
-                        properties["use_entities"] is None
-                        or properties["use_entities"] is False
-                    ):
-                        properties["use_entities"] = []
-            else:
-                name = intent
+            if not isinstance(intent, dict):
                 intent = {intent: {"use_entities": True, "ignore_entities": []}}
 
+            name, properties = list(intent.items())[0]
+
             if name in intent_properties.keys():
                 raise InvalidDomain(
                     "Intents are not unique! Found two intents with name '{}'. "
                     "Either rename or remove one of them.".format(name)
                 )
 
+            if (
+                properties.get("used_entities") is None
```

Comment:
Not on purpose. We should't introduce a third option, `used_entities` should be internal only.

The reason why I added handling for an existing property `used_entities` was that otherwise `self.__class_from_dict(self.as_dict())` wouldn't work: `from_dict` calls the domain constructor, so `collect_intent_properties` in this case is called with `used_entities` existing instead of `use_entities` or `ignore_entities`.

A possible solution to this:
How about I change the `as_dict` function to do the transformation back from `used_entities` to `use_entities`/`ignore_entities`?
Tobias pointed out that `as_dict` is only used for things that should go "outside" anyways, so if we want `used_entities` to be internal only changing it like this would be better anyways. Right now, e.g. in `server.py::get_domain` we give back the internal version with `used_properties`, but probably we shouldn't.
What do you think?

Reply:
yes, `get_domain()` should not return `used_entities` as this is returned to the user. making the transformation back in `as_dict()` makes sense